<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>bacterial_assembly.utf8</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioinformatics and Genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="labs.html">Lab Materials</a>
</li>
<li>
  <a href="syllabus.html">Syllabus</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="lab-9-bacterial-genome-assembly" class="section level2">
<h2>Lab 9: Bacterial Genome Assembly</h2>
<blockquote>
<p>Step 1: Launch an instance on Jetstream. For this exercise, we will use a m1.xlarge instance.</p>
</blockquote>
<blockquote>
<p>Update, upgrade, install…</p>
</blockquote>
<pre><code>sudo apt-get -y install build-essential python</code></pre>
<blockquote>
<p>Install Conda, then the following software..</p>
</blockquote>
<pre><code>canu quast spades</code></pre>
</div>
<div id="if-your-last-name-begins-with-a-l-do-this-section.-see-below-for-m-z" class="section level1">
<h1>If your last name begins with A-L, do this section. See below for M-Z</h1>
<blockquote>
<p>Download PacBio Data. This is 25x coverage of an E. coli genome.</p>
</blockquote>
<pre><code>cd
curl -L -o pacbio.fastq http://gembox.cbcb.umd.edu/mhap/raw/ecoli_p6_25x.filtered.fastq
curl -LO ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz</code></pre>
<blockquote>
<p>Assemble with canu (<a href="http://canu.readthedocs.io/en/latest/index.html" class="uri">http://canu.readthedocs.io/en/latest/index.html</a>). Canu is the leading long-read genome assembler as of 2017.</p>
</blockquote>
<pre><code>canu \
 -p ecoli -d ecoli-pacbio \
 genomeSize=4.8m \
 -pacbio-raw pacbio.fastq</code></pre>
<blockquote>
<p>compare the genomes, using quast</p>
</blockquote>
<pre><code>quast ecoli-pacbio/ecoli.contigs.fasta \
        -R GCF_000005845.2_ASM584v2_genomic.fna.gz \
        -o quast_output --threads 24 --gene-finding</code></pre>
</div>
<div id="last-names-m-z-do-this-section" class="section level1">
<h1>Last names M-Z, do this section</h1>
<blockquote>
<p>Download an Illumina dataset for E. coli, along with a reference genome</p>
</blockquote>
<pre><code>cd
curl -LO https://s3.amazonaws.com/gen711/ecoli_data.tar.gz
tar -zxf ecoli_data.tar.gz
curl -LO ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2/GCF_000005845.2_ASM584v2_genomic.fna.gz</code></pre>
<blockquote>
<p>Assemble with SPAdes (<a href="http://cab.spbu.ru/software/spades/" class="uri">http://cab.spbu.ru/software/spades/</a>), using a kmer length of 95. This is the genome assembler you will use, if you assemble Illumina data.</p>
</blockquote>
<pre><code> spades.py -t 24 -m 55 --mp1-rf -k 95 \
 --pe1-1 ecoli_pe.1.fq \
 --pe1-2 ecoli_pe.2.fq \
 --mp1-1 nextera.1.fq \
 --mp1-2 nextera.2.fq \
 -o Ecoli_all_data</code></pre>
<blockquote>
<p>compare the genomes, using quast</p>
</blockquote>
<pre><code>quast Ecoli_all_data/scaffolds.fasta \
        -R GCF_000005845.2_ASM584v2_genomic.fna.gz \
        -o quast_output --threads 24 --gene-finding</code></pre>
</div>
<div id="everybody-do-this-section" class="section level1">
<h1>Everybody do this section</h1>
<blockquote>
<p>look at report, found at <code>quast_output/report.txt</code>. How many pieces (contigs) is the genome in? How many should it be, ideally? How many pieces did the other strategy have as a result?</p>
</blockquote>
<blockquote>
<p>Download the full report, with pretty graphics, optional.</p>
</blockquote>
<pre><code>scp -ri jetkey username@xxx.xxx.xxx.x:quast_output ~/Downloads</code></pre>
<blockquote>
<p>Optional: Polish PacBio assembly using Quiver (<a href="https://github.com/PacificBiosciences/GenomicConsensus" class="uri">https://github.com/PacificBiosciences/GenomicConsensus</a>). This will take more time that we have in lab, but might be a good learning experience.</p>
</blockquote>
</div>
<div id="terminate-your-instance" class="section level1">
<h1>Terminate your instance</h1>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
