<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>khmer.utf8</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioinformatics and Genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="labs.html">Lab Materials</a>
</li>
<li>
  <a href="syllabus.html">Syllabus</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">




</div>


<div id="lab-7-khmer" class="section level2">
<h2>Lab 7: khmer</h2>
<p>During this lab, we will acquaint ourselves with digital normalization. You will:</p>
<ol style="list-style-type: decimal">
<li><p>Install software and download data</p></li>
<li><p>Quality and adapter trim data sets.</p></li>
<li><p>Apply digital normalization to the dataset.</p></li>
<li><p>Count and compare kmers and kmer distributions in the normalized and un-normalized dataset.</p></li>
<li><p>Plot in RStudio.</p></li>
</ol>
<p>The JellyFish manual: <a href="ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf" class="uri">ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf</a> The Khmer manual: <a href="http://khmer.readthedocs.org/en/v1.1" class="uri">http://khmer.readthedocs.org/en/v1.1</a></p>
<blockquote>
<p>Step 1: Launch, For this exercise, we will use a m1.large instance.</p>
</blockquote>
<blockquote>
<p>Install other software, including RStudio</p>
</blockquote>
<pre><code>sudo apt-get -y install build-essential python python-pip gdebi-core r-base
curl -LO  https://download2.rstudio.org/rstudio-server-1.0.143-amd64.deb
sudo gdebi -n rstudio-server-1.0.143-amd64.deb</code></pre>
<blockquote>
<p>Install Conda, and then the following software.</p>
</blockquote>
<pre><code>khmer jellyfish trimmomatic
</code></pre>
<blockquote>
<p>Download data</p>
</blockquote>
<pre><code>cd
curl -L https://s3.amazonaws.com/Mc_Transcriptome/Thomas_McBr1_R1.PF.fastq.gz &gt; kidney.1.fq.gz</code></pre>
<blockquote>
<p>Merge –&gt; Trim low quality bases and adapters from dataset –&gt; count kmers –&gt; make a histogram. Normalize in the 2nd command. Make sure you know what is going on here!</p>
</blockquote>
<p>::</p>
<pre><code>trim=2
norm=30

#paste the below lines together as 1 command


trimmomatic SE -threads 10 kidney.1.fq.gz /dev/stdout \
ILLUMINACLIP:$HOME/anaconda/install/envs/gen711/share/trimmomatic-0.38-1/adapters/TruSeq3-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 LEADING:2 TRAILING:2 MINLEN:25 \
| jellyfish count -m 25 -s 700M -t 8 -C -o /dev/stdout /dev/stdin \
| jellyfish histo /dev/stdin -o trimmed.no.normalize.histo

#and

#paste the below lines together as 1 command

trimmomatic SE -threads 10 kidney.1.fq.gz /dev/stdout \
ILLUMINACLIP:$HOME/anaconda/install/envs/gen711/share/trimmomatic-0.38-1/adapters/TruSeq3-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 LEADING:2 TRAILING:2 MINLEN:25 \
| normalize-by-median.py --max-memory-usage 1e9 -C $norm -o - - \
| jellyfish count -m 25 -s 700M -t 8 -C -o /dev/stdout /dev/stdin \
| jellyfish histo /dev/stdin -o trimmed.yes.normalize.histo</code></pre>
<blockquote>
<p>Now, you have 2 files: <code>trimmed.yes.normalize.histo</code> and <code>trimmed.no.normalize.histo</code>. these contain the kmer frequency data. The 1st column is the abundance, the 2nd column is the number of 25mers that have that abundance. Can you tell me how many unique kmers there are in each dataset? Does this make sense?</p>
</blockquote>
<blockquote>
<p>Launch RStudio, remember you have to make a new password, and find the web address. See lab 2 (blast lab) for details.</p>
</blockquote>
<pre><code>install.packages(&quot;beanplot&quot;)
library(&quot;beanplot&quot;)

#Import all 2 histogram datasets:


y_khmer &lt;- read.table(&quot;trimmed.yes.normalize.histo&quot;, quote=&quot;\&quot;&quot;)
n_khmer &lt;- read.table(&quot;trimmed.no.normalize.histo&quot;, quote=&quot;\&quot;&quot;)

# plot differences between non-unique kmers

par(mfrow=c(2,2))

plot(y_khmer$V2[0:300] - n_khmer$V2[0:300], type=&#39;l&#39;,
    xlim=c(0,300), xaxs=&quot;i&quot;, yaxs=&quot;i&quot;, frame.plot=F,
    ylim=c(-20000,20000), col=&#39;red&#39;, xlab=&#39;kmer frequency&#39;,
    lwd=4, ylab=&#39;count&#39;,
    main=&#39;Diff in 25mer counts of \n C20 normalized vs. un-normalized datasets&#39;)
abline(h=0)

plot(y_khmer$V2[0:100], type=&#39;l&#39;,
      xlim=c(0,100), xaxs=&quot;i&quot;, yaxs=&quot;i&quot;, frame.plot=F,
      ylim=c(0,180000), col=&#39;red&#39;, xlab=&#39;kmer frequency&#39;,
      lwd=4, ylab=&#39;count&#39;,
      main=&#39;Kmer distribution of C20-normalized dataset&#39;)

plot(n_khmer$V2[0:100], type=&#39;l&#39;,
      xlim=c(0,100), xaxs=&quot;i&quot;, yaxs=&quot;i&quot;, frame.plot=F,
      ylim=c(0,180000), col=&#39;red&#39;, xlab=&#39;kmer frequency&#39;,
      lwd=4, ylab=&#39;count&#39;,
      main=&#39;Kmer distribution of non-normalized dataset&#39;)
</code></pre>
<blockquote>
<p>What do the analyses of kmer counts tell you?</p>
</blockquote>
</div>
<div id="terminate-your-instance" class="section level2">
<h2>TERMINATE YOUR INSTANCE</h2>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
