<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>phylogenetics.utf8</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioinformatics and Genomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="labs.html">Lab Materials</a>
</li>
<li>
  <a href="syllabus.html">Syllabus</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">




</div>


<div id="lab-3-make-a-phylogeny" class="section level2">
<h2>Lab 3: Make a phylogeny!!!</h2>
<p>During this lab, we will acquaint ourselves with the steps to making a tree. Your objectives are:</p>
<ol style="list-style-type: decimal">
<li><p>Familiarize yourself with the software, how to execute it, how to visualize results.</p></li>
<li><p>Identity the unknown sequences.</p></li>
</ol>
<blockquote>
<p>Step 1: Launch an instance on Jetstream. For this exercise, we will use a m1.medium instance.</p>
</blockquote>
<pre><code>ssh -i jetkey username@xxx.xxx.xxx.xxx</code></pre>
<blockquote>
<p>The machine you are using is Linux Ubuntu: Ubuntu is an operating system you can use (I do) on your laptop or desktop. One of the nice things about this OS is the ability to update the software, easily.  The command <code>sudo apt-get update</code> checks a server for updates to existing software.</p>
</blockquote>
<pre><code>sudo apt-get update</code></pre>
<blockquote>
<p>The upgrade command actually installs any of the required updates.</p>
</blockquote>
<pre><code>sudo apt-get -y upgrade</code></pre>
<blockquote>
<p>OK, what are these commands?  <code>sudo</code> is the command that tells the computer that we have admin privileges. Try running the commands without the sudo – it will complain that you don’t have admin privileges or something like that. <em>Careful here, using sudo means that you can do something really bad to your own computer – like delete everything</em>, so use with caution. It’s not a big worry when using Jetstream, as this is a virtual machine- fixing your worst mistake is as easy as just terminating the instance and restarting.</p>
</blockquote>
<blockquote>
<p>So now that we have updates the software, lets see how to add new software. Same basic command, but instead of the <code>update</code> or <code>upgrade</code> command, we’re using <code>install</code>. EASY!!</p>
</blockquote>
<pre><code>sudo apt-get -y install ruby build-essential python python-pip gdebi-core r-base
</code></pre>
<blockquote>
<p>Install Conda. Conda is another package manager, but for scientific software. We will use it basically every week!</p>
</blockquote>
<pre><code>mkdir anaconda
cd anaconda
curl -LO https://repo.anaconda.com/archive/Anaconda3-5.1.0-Linux-x86_64.sh
bash Anaconda3-5.1.0-Linux-x86_64.sh -b -p $HOME/anaconda/install/
echo &quot;. $HOME/anaconda/install/etc/profile.d/conda.sh&quot; &gt;&gt; ~/.bashrc
source ~/.bashrc</code></pre>
<blockquote>
<p>Make a conda environment, activate it, and install raxml and mafft. Make sure you know what these software packages are used for.</p>
</blockquote>
<pre><code>conda create -y --name gen711
conda activate gen711
conda install -y -c bioconda raxml mafft</code></pre>
<blockquote>
<p>Download some data, using this command. The 1st dataset is data is from an unknown species, the 2nd is the UniProt database, which is a well curated set of protein sequences from many different organisms. It’s often a good 1st database to use for trying to identify sequences. Look at the 2 datasets using the comment <code>less</code>. Are these nucleotides or proteins?</p>
</blockquote>
<pre><code>cd $HOME
curl -LO https://s3.amazonaws.com/gen711/dataset1.pep
curl -LO https://s3.amazonaws.com/gen711/uniprot.pep
</code></pre>
<blockquote>
<p>In the interest of time, I am having you pull out the HOX genes from the Uniprot database, and 2 unknown sequences (ENSPTRP00000032491 and ENSPTRP00000032494). You’re goal is to tell me the identity of these unknowns using a tree.</p>
</blockquote>
<pre><code>grep --no-group-separator -A1 &#39;ENSPTRP00000032491\|ENSPTRP00000032494&#39; dataset1.pep &gt; query.pep
grep --no-group-separator -A1 &#39;HXA2_HUMAN\|HXA2_BOVIN\|HXA2_PAPAN\|HXA3_HUMAN\|HXA3_MOUSE\|HXA3_BOVIN\|HXA9_HUMAN&#39; uniprot.pep &gt; results.pep
cat query.pep results.pep &gt; for_alignment.pep</code></pre>
<blockquote>
<p>Align the proteins using mafft. It would be great if you figured out what the command is doing.</p>
</blockquote>
<pre><code>mafft --reorder --bl 80 --localpair --thread 6 for_alignment.pep &gt; for.tree</code></pre>
<blockquote>
<p>Make a phylogeny. It would be great if you figured out what the command is doing.</p>
</blockquote>
<pre><code>raxmlHPC-PTHREADS -f a -m PROTGAMMAAUTO -T 6 -x 37644 -N 100 -n tree -s for.tree -p 35 -o &quot;sp|P31269|HXA9_HUMAN&quot;</code></pre>
</div>
<div id="visualize-the-phylogeny" class="section level2">
<h2>Visualize the phylogeny…</h2>
<blockquote>
<p>Download and install RStudio</p>
</blockquote>
<pre><code>curl -LO  https://download2.rstudio.org/rstudio-server-1.1.456-amd64.deb
sudo gdebi -n rstudio-server-1.1.456-amd64.deb</code></pre>
<blockquote>
<p>Make a password for the Rstudio webserver (make it easy, like the word “password”!!!)</p>
</blockquote>
<pre><code>sudo passwd $(whoami)</code></pre>
<blockquote>
<p>Note that the text will not echo to the screen (because it’s a password!)</p>
</blockquote>
<blockquote>
<p>Find out the web address of your server. Paste the web address that comes up on the terminal, in to your browser.</p>
</blockquote>
<pre><code>printf &quot;\n\n\n\n The web address for my RStudio Web server, the one I need to paste into my browser, is:  http://$(hostname):8787/ \n\n\n\n&quot;</code></pre>
<div id="in-rstudio-not-the-terminal" class="section level3">
<h3>in RStudio (not the terminal)</h3>
<blockquote>
<p>Install packages.</p>
</blockquote>
<pre><code>install.packages(c(&quot;ape&quot;, &quot;ggplot2&quot;))
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(c(&quot;ggtree&quot;, &quot;Biostrings&quot;))

library(&quot;ape&quot;)
library(&quot;Biostrings&quot;)
library(&quot;ggplot2&quot;)
library(&quot;ggtree&quot;)</code></pre>
<blockquote>
<p>Read in the results file you just made and look at the results!!!</p>
</blockquote>
<pre><code>tree &lt;- read.raxml(&quot;RAxML_bipartitionsBranchLabels.tree&quot;)

ggtree(tree) + geom_label(aes(label=bootstrap, fill=bootstrap)) + geom_tiplab() +
scale_fill_continuous(low=&#39;darkgreen&#39;, high=&#39;red&#39;) + ggplot2::xlim(0, 4)</code></pre>
<p>Looking at the tree, can you tell me which type of HOX genes ENSPTRP00000032491 and ENSPTRP00000032494 are? If you can, you’ve just used comparative phylogenetics (instead of BLAST) to identify unknown sequences! Some might say this is a much more scientifically sound way of identifying unknown sequences. YAY!!!!</p>
</div>
</div>
<div id="terminate-your-instance" class="section level2">
<h2>Terminate your instance!!!</h2>
<p>This lab uses code from ANGUS2017: <a href="https://angus.readthedocs.io/en/2017/visualizing-blast-scores-with-RStudio.html" class="uri">https://angus.readthedocs.io/en/2017/visualizing-blast-scores-with-RStudio.html</a></p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
